File: backend/app/schemas/activity_log.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ActivityLogOut(BaseModel):
    id: uuid.UUID
    action: str
    entity_type: str
    entity_id: uuid.UUID
    details: dict | None
    ip_address: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class ActivityLogListResponse(BaseModel):
    items: list[ActivityLogOut]
    total: int
    page: int
    limit: int

File: frontend/src/types/activityLog.ts
export interface ActivityLogEntry {
  id: string
  action: string
  entity_type: string
  entity_id: string
  details: Record<string, unknown> | null
  ip_address: string | null
  created_at: string
}

export interface ActivityLogListResponse {
  items: ActivityLogEntry[]
  total: number
  page: number
  limit: number
}
